flow/table_test.go

```diff
@@ -0,0 +1,300 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package flow
+
+import (
+	"math/rand"
+	"net"
+	"testing"
+	"time"
+
+	"encoding/json"
+	j "github.com/gima/jsonv/src"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+func forgeEthIPTCP(t *testing.T, seed int64) *gopacket.Packet {
+	var options gopacket.SerializeOptions
+	rnd := rand.New(rand.NewSource(seed))
+
+	rawBytes := []byte{10, 20, 30}
+	ethernetLayer := &layers.Ethernet{
+		SrcMAC: net.HardwareAddr{0x00, 0x0F, 0xAA, 0xFA, 0xAA, byte(rnd.Intn(0x100))},
+		DstMAC: net.HardwareAddr{0x00, 0x0D, 0xBD, 0xBD, byte(rnd.Intn(0x100)), 0xBD},
+	}
+	ipLayer := &layers.IPv4{
+		SrcIP: net.IP{127, 0, 0, byte(rnd.Intn(0x100))},
+		DstIP: net.IP{byte(rnd.Intn(0x100)), 8, 8, 8},
+	}
+	tcpLayer := &layers.TCP{
+		SrcPort: layers.TCPPort(byte(rnd.Intn(0x10000))),
+		DstPort: layers.TCPPort(byte(rnd.Intn(0x10000))),
+	}
+	// And create the packet with the layers
+	buffer := gopacket.NewSerializeBuffer()
+	err := gopacket.SerializeLayers(buffer, options,
+		ethernetLayer,
+		ipLayer,
+		tcpLayer,
+		gopacket.Payload(rawBytes),
+	)
+	if err != nil {
+		t.Fail()
+	}
+
+	gpacket := gopacket.NewPacket(buffer.Bytes(), layers.LayerTypeEthernet, gopacket.Default)
+	return &gpacket
+
+}
+
+func generateFlows(t *testing.T, ft *FlowTable) []*Flow {
+	flows := []*Flow{}
+	for i := 0; i < 10; i++ {
+		packet := forgeEthIPTCP(t, int64(i))
+		flow, new := ft.GetFlow(string(i))
+		if !new {
+			t.Fail()
+		}
+		err := flow.fillFromGoPacket(packet)
+		if err != nil {
+			t.Error("fillFromGoPacket : " + err.Error())
+		}
+		flows = append(flows, flow)
+	}
+	return flows
+}
+
+func checkFlowTable(t *testing.T, ft *FlowTable) {
+	for uuid, f := range ft.table {
+		if uuid != f.UUID {
+			t.Error("FlowTable Collision ", uuid, f.UUID)
+		}
+	}
+}
+
+func NewFlowTableSimple(t *testing.T) *FlowTable {
+	ft := NewFlowTable()
+	var flows []*Flow
+	flow := &Flow{}
+	flow.UUID = "1234"
+	flows = append(flows, flow)
+	ft.Update(flows)
+	checkFlowTable(t, ft)
+	ft.Update(flows)
+	checkFlowTable(t, ft)
+	if "1 flows" != ft.String() {
+		t.Error("We should got only 1 flow")
+	}
+	flow = &Flow{}
+	flow.UUID = "4567"
+	flows = append(flows, flow)
+	ft.Update(flows)
+	checkFlowTable(t, ft)
+	ft.Update(flows)
+	checkFlowTable(t, ft)
+	if "2 flows" != ft.String() {
+		t.Error("We should got only 2 flows")
+	}
+	return ft
+}
+func NewFlowTableComplex(t *testing.T) *FlowTable {
+	ft := NewFlowTable()
+	flows := generateFlows(t, ft)
+	ft = ft.NewFlowTableFromFlows(flows)
+	checkFlowTable(t, ft)
+	return ft
+}
+
+func TestNewFlowTable(t *testing.T) {
+	ft := NewFlowTable()
+	if ft == nil {
+		t.Error("new FlowTable return null")
+	}
+}
+func TestFlowTable_String(t *testing.T) {
```

Comment:
I used a tool (go-tests) to generated the template
As String is the method for FlowTable, I think the _ make sense